### PR TITLE
improve ssh auth

### PR DESCRIPTION
**Pull request by leeif**
Changes:
1. enable network user and identity_file settings in yaml file.
2. provide an env variable [NETWORK_IDENTITY_FILE] to override the yaml identity_file setting in some use case like CI.
3. use go.mod to manage the dependance.
4. update crypto package to enable the openssh type key file parsing.

### CODE REVIEW

**Comment by leeif** [comment]
It seems I should also update the travis dot file

**Comment by leeif** [comment]
Oh, sorry. I will submit the issue instead.

File: ssh.go [hunk 1]
```diff
@@ -80,7 +81,7 @@ var initAuthMethodOnce sync.Once
 var authMethod ssh.AuthMethod
 
 // initAuthMethod initiates SSH authentication method.
-func initAuthMethod() {
+func initAuthMethod(identityFile string) {
```
Comment (SharkFourSix) [inline ssh.go:84]: Doesn't this introduce breaking changes? Better to refactor the code into another function with the identity file parameter and have the old one call the new one with ```null``` as argument to retain the default function's behavior.

```golang

func initAuthMethodWithIdentityFile(identityFile *string) {
    if identityFile != nil {
        // .. Load from identity file. Optionally checking the ENV for a decryption key as well.
    } else {
        // ... default behavior. load from agent etc
    }
}

func initAuthMethod(){
    initAuthMethodWithIdentityFile(nil);
}
```

File: ssh.go [hunk 2]
```diff
@@ -20,6 +20,7 @@ type SSHClient struct {
 	conn         *ssh.Client
 	sess         *ssh.Session
 	user         string
+	identityFile string
```
Comment (SharkFourSix) [inline ssh.go:23]: Also include option for looking up passphrase from the environment variables.

Use case:

Running sup from a github action, feeding a private key passphrase from the repository secrets, without the need to hardcode the ENV variable

**Pull request closed without merging**